src/Microsoft.AspNetCore.Rewrite/UrlRewrite/ServerVariables.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Text.RegularExpressions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Rewrite.UrlRewrite
+{
+    public static class ServerVariables
+    { 
+        public static Func<HttpContext, Match, Match, string> FindServerVariable(string serverVariable)
```

Comment:
For assembling the test strings. Instead of looking up the string and returning the appropriate part of the http context, we create a func that will just get it.